cpp/ql/src/Likely Bugs/Likely Typos/ExprHasNoEffect.ql

```diff
@@ -99,7 +99,7 @@ where // EQExprs are covered by CompareWhereAssignMeant.ql
       not peivc.(FunctionCall).getTarget().hasName("operator==") and
       not accessInInitOfForStmt(peivc) and
       not peivc.isCompilerGenerated() and
-      not exists(Macro m | peivc = m.getAnInvocation().getAnExpandedElement()) and
+      not peivc.isFromTemplateInstantiation(_) and
```

Comment:
Why remove the macro check? Is it redundant because `parent.isInMacroExpansion()` is checked below?

Reply:
Hmm, I didn't really intend to remove it.  I was curious whether that line affected any of the tests, and it didn't, but that doesn't necessarily mean it's truly doing nothing.  I'll add some test cases for the relationship between this and `parent.isInMacroExpansion()` and probably reinstate it.

Reply:
I've added a test case for macros and reinstated the macro check.  I now think it's actually the other macro clause that might be redundant - but with no explanatory comments or test cases, and few real world results that care, I'm not confident.